x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/user/User.java

```diff
@@ -49,7 +50,7 @@ public User(String username, String[] roles, String fullName, String email, Map<
 
     private User(String username, String[] roles, String fullName, String email, Map<String, Object> metadata, boolean enabled,
                 User authenticatedUser) {
-        this.username = username;
+        this.username = Objects.requireNonNull(username);
```

Comment:
nit: I hate when I get NPE in a line that has many calls and it's not quite obvious what threw it so I prefer to add a message to `requireNotNull`. Not sure if this is worth rerunning CI here though, just a suggestion

Reply:
But this line doesn't have many calls. As long as you can link the stacktrace to the code it's clear why this line throughs NPE.

That said, I'd be happy for the whole codebase to have a forbidden API on `requireNonNull` without a message, but we don't.